test/PerformanceTests/SerializationComparisonsTests/ExperimentTests/WriterTests.cs

```diff
@@ -0,0 +1,82 @@
+//---------------------------------------------------------------------
+// <copyright file="WriterTests.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using ExperimentsLib;
+using Xunit;
+
+namespace ExperimentTests
+{
+    public class WriterTests
+    {
+        public static IEnumerable<Customer> data = CustomerDataSet.GetCustomers(4);
+        public static WriterCollection<IEnumerable<Customer>> writers = DefaultWriterCollection.Create();
+        public static IEnumerable<object[]> WriterNames { get; } = writers.GetWriterNames()
+            .Where(n => !n.Contains("NoOp"))
+            .Select(n => new string[] { n });
+
+        public static IEnumerable<object[]> noOpWriterNames() =>
+            writers.GetWriterNames()
+            .Where(n => n.Contains("NoOp"))
+            .Select(n => new string[] { n });
+
+        [Theory]
+        [MemberData(nameof(WriterNames))]
+        public async Task WriterWritesCustomersCollectionPayload(string writerName)
+        {
+            using var stream = new MemoryStream();
+            var writer = writers.GetWriter(writerName);
+
+            await writer.WritePayload(data, stream);
+
+            using var expectedReader = new StreamReader("ExpectedOutput.txt");
+            string expectedOutput = expectedReader.ReadToEnd();

```

Comment:
There might be a case for reading from the stream asynchronously (`ReadToEndAsync`) since this is an async tests...

Reply:
Alright. Not sure if that affects the writer in any way though.